pac4j-core/src/main/java/org/pac4j/core/client/IndirectClient.java

```diff
@@ -40,7 +40,7 @@ protected void internalInit(final WebContext context) {
     }
 
     @Override
-    public final HttpAction redirect(final WebContext context) throws HttpAction {
+    public HttpAction redirect(final WebContext context) throws HttpAction {
```

Comment:
I'm not sure to follow you. Reading the source code again, I even think you could just override the `retrieveRedirectAction` method, the `redirect` and `getRedirectAction`  methods just wrap it with code you should not get rid of.